### PR TITLE
add nifty little "systemd-analyze calendar" command

### PR DESCRIPTION
**Pull request by poettering**
Writing calendar expressions is not always trivial. Let's add a little tool to test them with.

### CODE REVIEW

File: src/analyze/analyze.c [hunk 1]
```diff
@@ -1394,6 +1395,67 @@ static int dump_syscall_filters(char** names) {
...
+                }
+
+                if (!streq(t, *p))
+                        printf("  Original form: %s\n", *p);
+
+                printf("Normalized form: %s\n", t);
+
+                r = calendar_spec_next_usec(spec, n, &next);
+                if (r == -ENOENT)
+                        printf("Next elapsation: never\n");
```
Comment (keszybz) [inline src/analyze/analyze.c]: "elapse" please.

File: src/analyze/analyze.c [hunk 2]
```diff
@@ -1394,6 +1395,67 @@ static int dump_syscall_filters(char** names) {
...
+                else if (r < 0) {
+                        ret = log_error_errno(r, "Failed to determine next event for '%s': %m", *p);
```
Comment (keszybz) [inline src/analyze/analyze.c]: "next elapse"

File: src/analyze/analyze.c [hunk 3]
```diff
@@ -1394,6 +1395,67 @@ static int dump_syscall_filters(char** names) {
...
+                        continue;
+                } else {
+                        char buffer[CONST_MAX(FORMAT_TIMESTAMP_MAX, FORMAT_TIMESTAMP_RELATIVE_MAX)];
+
+                        printf("Next elapsation: %s\n", format_timestamp(buffer, sizeof(buffer), next));
```
Comment (keszybz) [inline src/analyze/analyze.c]: "elapse"

File: man/systemd-analyze.xml [hunk 4]
```diff
@@ -123,6 +123,12 @@
       <arg choice="plain">verify</arg>
       <arg choice="opt" rep="repeat"><replaceable>FILES</replaceable></arg>
     </cmdsynopsis>
+    <cmdsynopsis>
+      <command>systemd-analyze</command>
+      <arg choice="opt" rep="repeat">OPTIONS</arg>
+      <arg choice="plain">calendar</arg>
+      <arg choice="opt" rep="repeat"><replaceable>SPECS</replaceable></arg>
```
Comment (keszybz) [inline man/systemd-analyze.xml]: Drop `choice="opt"`, this command does not work without at least one spec.

File: man/systemd.time.xml [hunk 5]
```diff
@@ -300,6 +300,10 @@ Wed..Sat,Tue 12-10-15 1:2:3 → Tue..Sat 2012-10-15 01:02:03
       <citerefentry><refentrytitle>systemd.timer</refentrytitle><manvolnum>5</manvolnum></citerefentry>
       for details.</para>
 
+      <para>Use the <command>calendar</command> command of
+      <citerefentry><refentrytitle>systemd-analyze</refentrytitle><manvolnum>1</manvolnum></citerefentry> to validate
+      and normalize calendar time specifications for testing purpose. The tool also calculates when a specified
```
Comment (fsateler) [inline man/systemd.time.xml]: "purposes"

**Comment by poettering** [comment]
Force pushed a new version now, with all points addressed. I also made two minor additional changes:

1. I tweaked the output a bit, so that when the local timezone is set to UTC we don't unnecessarily output the UTC timestamp twice.

2. I replaced elapsation → elapsation in two other places in the tree as well.

PTAL

**Pull request merged**